Specification.md

```diff
@@ -78,52 +104,52 @@ Defined via a [transcript](#transcript):
-The first sidechain block contains no transactions, empty state and has height `...` (TBD).
-
-
 #### Reward
 
 Reward is an asset with flavor `0` that’s given to the creators of the [block](#block) in proportion to their [bids](#bid). 
 
-Reward consists of [transaction fees](#fee) and [inflation](#inflation). 
+Reward consists of [transaction fees](#fee) and [inflation](#inflation).
 
 For each transaction, the fee is distributed among bids that include that transaction.
 
 Inflation is distributed among all the bids.
 
+Reward for the bag computes after [late period](#late-period). In other words, reward for the block with height `H` will be computed when a block with height `H+L` is arrived.
+
 For a block with a given height `h` the reward is computed as follows:
 
-1. Sum up all bid amounts (in satoshis) into `X`.
-2. For each transaction `tx_k`, sum up bid amounts from the bags that contain that transaction into `Z_k`.
-3. For each bid with amount `x_i` and transaction `tx_k`, compute the fee reward `F_{i,k} = fee * x_i / Z_k` (128-bit division of 64-bit unsigned integers, rounding down).
-4. Compute the inflation amount `R` according to the block height:
+1. For each bid `i` compute share:
+    1. Initial share `x` is a bid amount (in satoshis).
+    2. Find a block in which the bag with that bid is directly connected. Let it be at height `h'`.
+    3. Calculate discount for the bid: `d = (h - h' - 1) / L`, where `L` is the [late period](#late-period). Note that in case when `h - h' > L` chain is invalid, so we don't consider this case.
+    4. Multiply share at discount: `x = x - (x * d)`.
+2. Sum up all bid shares into `X`.
+3. Sum up all transaction fees into `F`. If the transaction is contained in more than 1 bag, add it only once.
+4. For each bid with share `x_i`, compute the fee reward `F_i = x_i * F / X` (128-bit division of 64-bit unsigned integers, rounding down).
+5. Compute the inflation amount `R` according to the block height:
     1. Subtract the initial block [height](#height): `h' = h - INITIAL_HEIGHT`
     2. Start with inflation `R = 50'000'000` and while the `h' > 210'000` and inflation is greater than zero, divide the inflation by 2 (rounding down) and subtract 210'000 from `h'`.
-    3. For each bid with amount `x_i` compute inflation reward `R_i = R * x_i / X` (128-bit division of 64-bit unsigned integers, rounding down).
-5. For each bid `i`, sum up `F_{i,k}` from each transaction `tx_k` and add inflation reward `R_i`. The resulting amount is the total award `T_i` per bid. 
-6. For each bid `bid_i` and total award `T_i` create a UTXO with the predicate `bid_i.address`. Unique anchor is computed as follows using the [transcript](#transcript): 
+    3. For each bid with share `x_i` compute inflation reward `R_i = R * x_i / X` (128-bit division of 64-bit unsigned integers, rounding down).
+6. For each bid `i`, sum up fee reward `F_i` and inflation reward `R_i`. The resulting amount is the total award `T_i` per bid. 
```

Comment:
it's simpler and less accumulated error to sum up all reward first (inflation + fees), then divide it among the minters.

Reply:
Fixed in 37da367